### PR TITLE
add new media type application/vnd.api+json

### PR DESCRIPTION
**Pull request by Doikor**
see https://www.iana.org/assignments/media-types/application/vnd.api+json for its specification

### CODE REVIEW

File: core/src/main/scala/org/http4s/MediaType.scala [hunk 1]
```diff
@@ -260,6 +260,7 @@ object MediaType extends Registry {
   val `application/postscript` = app("postscript", compressible, binary, "ai", "eps", "ps")
   val `application/rss+xml` = app("rss+xml", compressible, notBinary, "rss")
   val `application/soap+xml` = app("soap+xml", compressible, notBinary)
+  val `application/vnd.api+json` = app("vnd.api+json", compressible, binary) // we treat JSON as binary, since it's encoding is not variable but defined by RFC4627
```
Comment (ChristopherDavenport) [inline core/src/main/scala/org/http4s/MediaType.scala]: Does this need to be added to the types in logger that actually gets parsed as text despite being binary?

**Comment by Doikor** [comment]
@rossabaker I guess I'll fix the typos now as I'm changing the file.

File: core/src/main/scala/org/http4s/MediaType.scala (hunk 1 above)
Comment (Doikor) [reply core/src/main/scala/org/http4s/MediaType.scala]: Not really sure. Just doing it the same as the other json media types.

**Comment by rossabaker** [comment]
Appveyor failure is spurious.  Merging.

**Pull request merged**